lib/plug/adapters/cowboy2.ex

```diff
@@ -0,0 +1,306 @@
+defmodule Plug.Adapters.Cowboy2 do
+  @moduledoc """
+  Adapter interface to the Cowboy2 webserver.
+
+  ## Options
+
+    * `:ip` - the ip to bind the server to.
+      Must be either a tuple in the format `{a, b, c, d}` with each value in `0..255` for IPv4
+      or a tuple in the format `{a, b, c, d, e, f, g, h}` with each value in `0..65535` for IPv6.
+
+    * `:port` - the port to run the server.
+      Defaults to 4000 (http) and 4040 (https).
+
+    * `:acceptors` - the number of acceptors for the listener.
+      Defaults to 100.
+
+    * `:max_connections` - max number of connections supported.
+      Defaults to `16_384`.
+
+    * `:dispatch` - manually configure Cowboy's dispatch.
+      If this option is used, the given plug won't be initialized
+      nor dispatched to (and doing so becomes the user's responsibility).
+
+    * `:ref` - the reference name to be used.
+      Defaults to `plug.HTTP` (http) and `plug.HTTPS` (https).
+      This is the value that needs to be given on shutdown.
+
+    * `:compress` - Cowboy will attempt to compress the response body.
+      Defaults to false.
+
+    * `:timeout` - Time in ms with no requests before Cowboy closes the connection.
+      Defaults to 5000ms.
+
+    * `:protocol_options` - Specifies remaining protocol options,
+      see [Cowboy2 protocol docs](http://ninenines.eu/docs/en/cowboy/2.0/manual/cowboy_protocol/).
```

Comment:
The link is broken.

Reply:
Cowboy docs for v2.0 states that many options are shared and I can't find a separate `cowboy_protocol` document for v2.0.

Reply:
Good spot. https://ninenines.eu/docs/en/cowboy/2.0/manual/cowboy_http/ is probably the closest